versionology/has-version.go

```diff
@@ -0,0 +1,27 @@
+package versionology
+
+import "github.com/Masterminds/semver/v3"
+
+type HasVersion interface {
+	GetVersion() *semver.Version
+}
```

Comment:
Sorry if this is a nitpick, but `HasVersion` as the name of an interface with a `GetVersion` function is extremely confusing to me.

Reply:
`VersionFetcher`?

Reply:
I originally preferred `HasVersion` to have a function `Version()` but that didn't work with having a field `version` so I changed the function to `GetVersion()`. I'm adding more comments to the `versionology` package so hopefully it will be more clear.

Reply:
That being said, looks like a field `version` and function `Version()` should be fine... I thought I tried this though.

https://go.dev/doc/effective_go#Getters

> Go doesn't provide automatic support for getters and setters. There's nothing wrong with providing getters and setters yourself, and it's often appropriate to do so, but it's neither idiomatic nor necessary to put Get into the getter's name. If you have a field called owner (lower case, unexported), the getter method should be called Owner (upper case, exported), not GetOwner. The use of upper-case names for export provides the hook to discriminate the field from the method. A setter function, if needed, will likely be called SetOwner. Both names read well in practice: